hydra-cluster/src/CardanoNode.hs

```diff
@@ -125,6 +125,28 @@ getCardanoNodeVersion :: IO String
 getCardanoNodeVersion =
   readProcess "cardano-node" ["--version"] ""
 
+-- | Tries to reach for an existing cardano-node running on default state directory.
+findRunningCardanoNode :: FilePath -> KnownNetwork -> IO (Maybe RunningNode)
+findRunningCardanoNode stateDirectory knownNetwork = do
+  let args'@CardanoNodeArgs{nodeSocket} = defaultCardanoNodeArgs
+  let nodeSocketPath = File nodeSocket
+  maybeSocketPath <- doesFileExist $ unFile nodeSocketPath
+  if maybeSocketPath
+    then do
```

Comment:
Could: not do the `doesFileExist` at all.

The `try` below will still yield a `Nothing` if the file does not exist. After all, the notion of file existing here is worthless because it can also not exist when you try access it below.

This is "ask for forgiving, instead of allowance".